packages/Treasury/src/Services/TreasuryAnalyticsService.php

```
@@ -0,0 +1,214 @@
+        private InvestmentQueryInterface $investmentQuery,
+        private TreasuryPositionService $positionService,
+        private ?PayableDataProviderInterface $payableProvider = null,
+        private ?ReceivableDataProviderInterface $receivableProvider = null,
+        private LoggerInterface $logger = new NullLogger()
+    ) {}
+
+    public function calculateKPIs(
+        string $tenantId,
+        ?DateTimeImmutable $asOfDate = null
+    ): TreasuryAnalyticsInterface {
+        $date = $asOfDate ?? new DateTimeImmutable();
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$date'. (undefined)

(UnusedLocalVariable)